libraries/AP_ESC_Telem/AP_ESC_Telem.h

```diff
@@ -108,6 +115,12 @@ class AP_ESC_Telem {
     uint32_t _last_rpm_log_us[ESC_TELEM_MAX_ESCS];
     uint8_t next_idx;
 
+#if AP_SCRIPTING_ENABLED
+    // allow for scaling of RPMs via lua scripts
```

Comment:
This is always related to the number of poles via ```erpm * (200 / _bdshot.motor_poles)```. I think a much better change is to record whether an ESC is recording in rpm or erpm and then make the appropriate calculation.

Reply:
how would we record that, noting that it could be different for each ESC